ast-nodes/routine-declaration.dart

```diff
@@ -145,6 +145,11 @@ class RoutineDeclaration extends Declaration implements ScopeCreator {
       )
     );
 
+    for (var i = 0; i < this.parameters.length; i++) {
+      parameters[i].scopeMark.resolve(parameters[i].name).opaqueValue =
+          llvm.LLVMGetParam(module.getLastRoutine(), i);
```

Comment:
You have the reference to the routine in the `routine` variable, no need to fetch it from the module.

Reply:
Also, while you're going through parameters, it would be nice to call `LLVMSetValueName2(value, MemoryManager.getCString(parameters[i].name), parameters[i].name.length)`, where `value` is the thing you're getting from LLVMGetParam.
This will produce IR that is easier to read because we'll know the names of arguments